connector/src/main/java/org/jboss/as/connector/subsystems/datasources/GetDataSourceClassInfoOperationHandler.java

```diff
@@ -0,0 +1,235 @@
+                throw new OperationFailedException(ConnectorLogger.SUBSYSTEM_DATASOURCES_LOGGER.failedToLoadModuleDriver(mid
+                        .toString()), e);
+            }
+        }
+        if (cl == null) {
+            cl = getClassLoader(GetDataSourceClassInfoOperationHandler.class);
+        }
+        try {
+            cls = Class.forName(clsName, true, cl);
+        } catch (ClassNotFoundException e) {
+        }
+        if (cls == null) {
+            throw ConnectorLogger.SUBSYSTEM_DATASOURCES_LOGGER.failedToLoadDataSourceClass(clsName);
+        }
+        LinkedHashMap<String, Type> result = new LinkedHashMap<>();
+        for (Method method : allPublicSetterMethods(cls)) {
+            String mtdName = method.getName();
+            result.putIfAbsent(deCapitalize(mtdName.substring(3)), method.getParameterTypes()[0]);
+        }
+        return result.entrySet().stream().sorted((e1, e2) -> e1.getKey().compareTo(e2.getKey()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1, LinkedHashMap::new));
+    }
+
+    /**
+     * Check whether the types that JCA Injection knows.
+     *
+     * @see Injection.findMethod()
+     * @param clz the class
+     * @return whether it is understandable
+     */
+    private static boolean isTypeMatched(Class<?> clz) {
+        if (clz.equals(String.class)) {
+            return true;
+        } else if (clz.equals(byte.class) || clz.equals(Byte.class)) {
+            return true;
+        } else if (clz.equals(short.class) || clz.equals(Short.class)) {
+            return true;
+        } else if (clz.equals(int.class) || clz.equals(Integer.class)) {
+            return true;
+        } else if (clz.equals(long.class) || clz.equals(Long.class)) {
+            return true;
+        } else if (clz.equals(float.class) || clz.equals(Float.class)) {
+            return true;
+        } else if (clz.equals(double.class) || clz.equals(Double.class)) {
+            return true;
+        } else if (clz.equals(boolean.class) || clz.equals(Boolean.class)) {
+            return true;
+        } else if (clz.equals(char.class) || clz.equals(Character.class)) {
+            return true;
+        } else if (clz.equals(InetAddress.class)) {
+            return true;
+        } else if (clz.equals(Class.class)) {
+            return true;
+        } else if (clz.equals(Properties.class)) {
+            return true;
+        }
+        return false;
+    }
+
+    private static String deCapitalize(String str) {
+        if (str.length() == 1) {
+            return str.toLowerCase(Locale.US);
+        }
+        if (str.equals(str.toUpperCase(Locale.US))) { // all uppercase, just return
+            return str;
+        }
+        return str.substring(0, 1).toLowerCase(Locale.US) + str.substring(1);
+    }
+
+    private static ClassLoader getClassLoader(final Class<?> clz) {
+        if (System.getSecurityManager() == null)
```

Comment:
OperationStepHandlers are privilidged code, they should never have user code on the stack so don't need Privileged blocks